src/components/App/index.tsx

```diff
@@ -51,47 +40,32 @@ class AppBase extends React.Component<Props, State> {
   }
 
   async componentDidUpdate(prevProps: Props) {
-    const { apiState } = this.props;
+    const { apiState, dispatch, profile } = this.props;
 
-    if (
-      !this.state.profile &&
-      prevProps.apiState.authToken !== apiState.authToken
-    ) {
-      const profile = (await api.callApi({
+    if (!profile && prevProps.apiState.authToken !== apiState.authToken) {
+      const response = (await api.callApi({
         apiState,
         endpoint: '/accounts/profile/',
-      })) as State['profile'];
+      })) as ExternalUser;
 
-      if (profile && profile.name) {
-        // eslint-disable-next-line react/no-did-update-set-state
-        this.setState({ profile });
+      if (response && response.name) {
```

Comment:
I suppose we should look for `response.error` and deal with that too.

Reply:
Well, I think this is temporary code, so I'm not sure we need to make it perfect at this point. Also, What we're doing here is setting the user to "logged in" if the response contains a `name`. Wouldn't `name` be missing on an error, and therefore our code would work and not log the user in? We'd obviously want some messaging around a failed log in like that, but I'm not sure we should add that now, when I imagine this changing in the future.

Reply:
That's correct, and I also think we'll iterate over this code again (i.e. no need to apply changes here)

I was thinking that maybe it would make more sense to test `response.error` than `response.name` but 🤷‍♂️

Reply:
Ah, I didn't realize that's what you were suggesting. Is it okay to just leave it as is for now?

Reply:
yep. (and no, I was not suggesting anything initially, I was thinking out loud)